src/modalities/utils/number_conversion.py

```diff
@@ -98,6 +103,20 @@ def get_local_num_batches_from_num_samples(
         """
         return (global_num_samples) // num_ranks // local_micro_batch_size
 
+    @staticmethod
+    def get_num_samples_from_num_tokens(num_tokens: int, sequence_length: int) -> int:
+        """Calculates the number of samples given the global number of tokens and sequence length.
+
+        Args:
+            num_tokens (int): Global number of tokens.
+            sequence_length (int): Sequence length of the model.
+
+        Returns:
+            int: Number of samples.
+        """
+        num_samples = num_tokens // sequence_length
```

Comment:
Add warning if num_tokens is not divisible by sequence_length?

Reply:
In all other cases, in which elements  are cut off due to integer division we also don't have a warning. 
However, when we run modalities, we print out information on how many tokens we are effectively going to use and how many we are missing. This should be sufficient, no?